articles/budougumi0617.re

```diff
@@ -1,18 +1,282 @@
-= 動作確認用
+= 準標準パッケージ（golang.org/x）の早めぐり
 
 #@# textlint-disable
 
-## はじめに
-
-freee株式会社でバックエンドエンジニアをしている@budougumi0617@<fn>{bd617_twitter}です。
+== はじめに
 
 #@# textlint-enable
 
+
+freee株式会社でバックエンドエンジニアをしている@<code>{@budougumi0617}@<fn>{bd617_twitter}です。
+本章では2019年9月現在@<code>{golang.org/x}配下にある準公式パッケージについて紹介したいと思います。
+
+
+
 //footnote[bd617_twitter][@<href>{https://twitter.com/budougumi0617}]
 
-=== 本章の目的・ゴール
-このファイルはreviewdogの動作確認用です。
-きっと長い文書の中にできるだけ「の」を使ったりすればreviewdogの設定で呼ばれているtexlintのワーニングが呼び出されるはず。
+== 本章の概要・ゴール
+本章では、Goの準公式パッケージとして配信されているリポジトリにどんなものリポジトリがあるのかを紹介します。
+リポジトリの中にはGoの今後のバージョンのために試験的に作成されたパッケージから、実務での利用に耐えうるパッケージまでさまざまなコードが含まれています。
+中には実際に実務でGoを開発する際にほぼ確実に利用するであろうコマンドラインツールも含まれています。
+このような準硬式パッケージを学習することで、本章では次のようなことを学ぶことをゴールとします。
+
+ * 準公式パッケージで提供されるコマンドラインツールを知ることで日々の開発の生産性を上げる
+ * 準公式パッケージで提供されるコードを利用することでコーディングの品質を上げる（車輪の再発明を避ける）
+ * 今後リリースされるGoのバージョンに取り込まれるであろう、新しいGoの仕組みを先取りして学習する
+
+
+
+== @<code>{golang.org/x}配下にあるパッケージとは
+@<code>{golang/x}パッケージという準公式パッケージがあることをご存知でしょうか。
+例を挙げると、Go1.7で@<code>{context}パッケージが追加される前は@<code>{golang.org/x/net/context}パッケージの@<code>{context.Context}でコンテキスト情報を扱っていました。
+最近ではGo Modules導入前の@<code>{vgo}、まだ@<code>{xerrors}なども準公式パッケージです。
+準公式パッケージは公式パッケージに比べて互換性（@<i>{compatibility requirements}）の基準が低いなどの制約はありますが、便利なものばかりです。
+
+TODO: もうちょっと練って書く
+
+== @<code>{golang.org/x}で提供されているパッケージ一覧
+それでは、早速@<code>{golang.org/x}から提供されているパッケージを見ていきましょう。
+サブパッケージまでひとつひとつ紹介していくと途方のない数になってしまうため、ここでは@@<i>{GitHub}上のリポジトリ単位で紹介していきます。
+
+
+=== @<code>{golang.org/x/arch}パッケージ
+ * https://godoc.org/golang.org/x/arch
+ * https://github.com/golang/arch
+
+マシンアーキテクチャ依存のコードが同梱されている。Goのツールチェインで使われている。
+
+=== @<code>{golang.org/x/benchmarks}パッケージ
+ * https://godoc.org/golang.org/x/benchmarks
+ * https://github.com/golang/benchmarks
+
+
+ベンチマークを実行するツールがいくつか含まれているリポジトリ。
+ベンチマークの流れはドライバーで実装されている。ツール作る際の共通化の参考になりそう
+https://github.com/golang/benchmarks/blob/master//driver/driver.go
+
+GCについての知見？
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Disallow to use "？". (ja-technical-writing/no-exclamation-question-mark)